topology/probes/k8s/networkpolicy.go

```diff
@@ -77,7 +77,7 @@ func (n *networkPolicyProbe) OnAdd(obj interface{}) {
 	if np, ok := obj.(*v1beta1.NetworkPolicy); ok {
 		logging.GetLogger().Debugf("Adding %s", dumpNetworkPolicy(np))
 		n.graph.Lock()
-		npNode := newNode(n.graph, networkPolicyUID(np), n.newMetadata(np))
+		npNode := NewNode(n.graph, networkPolicyUID(np), n.newMetadata(np))
```

Comment:
so you want to externalize the your newNode(g,a,b) equivalent to n.graph.NewNode(a,b) 
to NewNode(n.graph,a,b)  equivalent to n.graph.NewNode(a,b)  ?

Do see the problem here ?
(same for addMetadata ...)

Reply:
moved code back to k8s/probes.go ... the NewNode is the generic function also used by istio, newNode is implemented in each k8s and istio and sets the correct manager (either "k8s" or "istio")

Reply:
in k8s probe

func NewNode(g *graph.Graph, i graph.Identifier, m graph.Metadata) *graph.Node {
	return g.NewNode(i, m, hostID)
}

where is the gain/enhancement ?
 1/ there are no value added
 2/ it's introduce confusion during rereading / debugging
 3/ you pollute the k8s namespace by exporting these functions

In this case : 
n.graph.NewNode(networkPolicyUID(np), n.newMetadata(np))

will do the same thing

Reply:
setting the correct `hostID` is the reason for this function

Reply:
Thinking out loud : now that we have an "Origin" field and with the PR (https://github.com/skydive-project/skydive/pull/1210), maybe we could set the `host` attribute of the analyzer graph to the empty string.
This way, you won't have to explicitely specify an empty host.

Reply:
@lebauce / @nplanel - I am not changing the current implementation - only making the existing functions public (and adding the manager field). I am fine with any such change but would like this to be done as a separate PR.